Userland/DevTools/HackStudio/Git/DiffViewer.cpp

```diff
@@ -39,11 +39,18 @@ void DiffViewer::paint_event(GUI::PaintEvent& event)
     size_t y_offset = 10;
     size_t current_original_line_index = 0;
     for (const auto& hunk : m_hunks) {
-        for (size_t i = current_original_line_index; i < hunk.original_start_line; ++i) {
-            draw_line(painter, m_original_lines[i], y_offset, LinePosition::Both, LineType::Normal);
-            y_offset += line_height();
+        // If there is no original target start line, this file is new to the git repository, attempting to loop over the lines will result in a crash.
+        if (hunk.original_start_line == (size_t)-1) {
```

Comment:
Where does the `-1` come from? That's a very unfortunate API. Can we make this use `Optional<size_t>` instead?

Reply:
It appears that LibDiff initializes ``HunkLocation.original_start_line`` with -1 in certain cases, I will look into this, and see if we can return an optional instead.